src/main/java/sg/ncl/MainController.java

```diff
@@ -2744,6 +2747,105 @@ public String adminTeamUsage(Model model,
+            return "redirect:/energy_usage";
+        }
+
+        String responseBody = responseEntity.getBody().toString();
+        JSONArray jsonArray = new JSONArray(responseBody);
+
+        // handling exceptions from SIO
+        if (RestUtil.isError(responseEntity.getStatusCode())) {
+            MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+            ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+            switch (exceptionState) {
+                case START_DATE_AFTER_END_DATE_EXCEPTION:
+                    log.warn("Get energy usage : Start date after end date error");
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERR_START_DATE_AFTER_END_DATE);
+                    return "redirect:/energy_usage";
+                default:
+                    log.warn("Get energy usage : sio or deterlab adapter connection error");
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+                    return "redirect:/energy_usage";
+            }
+        } else {
+            log.info("Get energy usage info : {}", responseBody);
+        }
+
+        DecimalFormat df2 = new DecimalFormat(".##");
+
+        double sumEnergy = 0.00;
+        List<String> listOfDate = new ArrayList<String>();
+        List<Double> listOfEnergy = new ArrayList<Double>();
+        ZonedDateTime currentZonedDateTime = convertToZonedDateTime(start);
+        String currentDate = null;
+        for (int i = 0; i < jsonArray.length(); i++) {
+            sumEnergy  += jsonArray.getDouble(i);
+
+            // add into listOfDate to display graph
+            currentDate = currentZonedDateTime.format(formatter);
+            listOfDate.add(currentDate);
+
+            // add into listOfEnergy to display graph
+            double energy = Double.valueOf(df2.format(jsonArray.getDouble(i)));
+            listOfEnergy.add(energy);
+
+            currentZonedDateTime = convertToZonedDateTime(currentDate).plusDays(1);
+        }
+
+        sumEnergy = Double.valueOf(df2.format(sumEnergy));
+        model.addAttribute("listOfDate", listOfDate);
+        model.addAttribute("listOfEnergy", listOfEnergy);
+        model.addAttribute("energy", sumEnergy);
+        return "/energy_usage";
+    }
+
+    /**
+     * Get simple ZonedDateTime from date string in the format 'YYYY-MM-DD'.
+     * @param date  date string to convert
+     * @return      ZonedDateTime of
+     */
+    private ZonedDateTime convertToZonedDateTime(String date) {
+        if (date != null) {
+            String[] result = date.split("-");
+            return ZonedDateTime.of(
+                    Integer.parseInt(result[0]),
+                    Integer.parseInt(result[1]),
+                    Integer.parseInt(result[2]),
+                    0, 0, 0, 0, ZoneId.of("Asia/Singapore"));
+        }
+        return null;
+    }
+
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'convertToZonedDateTime' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2259)